server/apps/publish/commands.py

```diff
@@ -56,4 +62,126 @@ def update_overdue_scheduled(self):
+                        takes_package_id = takes_package_service.get_take_package_id(item)
+                        if takes_package_id:
+                            removed_items.add(takes_package_id)
+                            takes_package = published_service.find_one(req=None, item_id=takes_package_id)
+                            if takes_package.get('expiry') > expiry_datetime:
+                                logger.info('Digital Story: {} isn''t expired'.format(takes_package_id))
+                            elif self._are_takes_expired(takes_package, to_remove, removed_items, expiry_datetime,
+                                                         item_id):
+                                to_remove.append(takes_package_id)
+                        else:
+                            to_remove.append(item_id)
+
+                    for item_id in to_remove:
+                        logger.info('Removing killed item with id: {} and slugged: {} from published collection'
+                                    .format(item_id, item.get('slugline')))
+                        get_resource_service('published').delete({'item_id': item_id})
+
+    def _are_takes_expired(self, takes_package, to_remove, removed_items, expiry_datetime, item_id=None):
+        """
+        Checks if all the takes in the given takes_package are expired. Each expired article is appended to the given
+        to_remove and removed_items collections. Returns True if all of them are expired, otherwise False.
+
+        :param takes_package:
+        :type takes_package: dict
+        :param to_remove:
+        :type to_remove: list
+        :param removed_items:
+        :type removed_items: set
+        :param expiry_datetime: timestamp to be used to check if a take is expired or not
+        :type expiry_datetime: datetime
+        :param item_id:
+        :type item_id:
+        :return: True if all of them are expired, otherwise False.
+        :rtype: bool
+        """
+
+        published_service = get_resource_service('published')
+        takes_package_service = TakesPackageService()
+
+        for takes_ref in takes_package_service.get_package_refs(takes_package):
+            if takes_ref[RESIDREF] != item_id:
+                removed_items.add(takes_ref[RESIDREF])
+                take = published_service.find_one(req=None, item_id=takes_ref[RESIDREF])
+                if take.get('expiry') < expiry_datetime:
+                    to_remove.append(takes_ref[RESIDREF])
+                else:
+                    logger.info('Take: {} isn''t expired'.format(takes_ref[RESIDREF]))
+                    to_remove.clear()
+                    return False
+
+        if item_id:
+            to_remove.append(item_id)
+
+        return True
+
+
 superdesk.command('publish:remove_overdue_scheduled', UpdateOverdueScheduledPublishedContent())
+superdesk.command('published:remove_killed_if_expired', UpdateOverdueScheduledPublishedContent())
```

Comment:
Did you meant `RemoveExpiredKilledContent` instead of `UpdateOverdueScheduledPublishedContent`?